tests/cli/test_codegen.py

```diff
@@ -80,6 +124,87 @@ def test_codegen(cli_runner: CliRunner, query_file_path: Path, tmp_path: Path):
     assert code_path.read_text() == "# This is a test file for GetUser"
 
 
+def test_codegen_injection_parameters(
+    cli_runner: CliRunner, query_file_path: Path, tmp_path: Path
+):
+    selector = "tests.fixtures.sample_package.sample_module:schema"
+    result = cli_runner.invoke(
+        app,
+        [
+            "codegen",
+            "-p",
+            "tests.cli.test_codegen:QueryCodegenTestPluginWithInjectableQuery",
+            "-o",
+            str(tmp_path),
+            "--schema",
+            selector,
+            str(query_file_path),
+        ],
+    )
+
+    assert result.exit_code == 0
+
+    code_path = tmp_path / "test.py"
+
+    assert code_path.exists()
+    assert code_path.read_text() == "# This is a test file for GetUser"
+
+
+def test_codegen_multiple_files(
+    cli_runner: CliRunner, query_file_path: Path, query_file_path2: Path, tmp_path: Path
+):
+    expected_paths = [
+        tmp_path / "query.py",
+        tmp_path / "query2.py",
+        tmp_path / "query.ts",
+        tmp_path / "query2.ts",
+    ]
+    for path in expected_paths:
+        assert not path.exists()
+
+    selector = "tests.fixtures.sample_package.sample_module:schema"
+    result = cli_runner.invoke(
+        app,
+        [
+            "codegen",
+            "-p",
+            "python",
+            "-p",
+            "typescript",
+            "-o",
+            str(tmp_path),
+            "--schema",
+            selector,
+            str(query_file_path),
+            str(query_file_path2),
+        ],
+    )
+
+    assert result.exit_code == 0
+
+    for path in expected_paths:
+        assert path.exists()
+        assert " GetUserResult" in path.read_text()
+
+
+def test_codegen_pass_no_query(cli_runner: CliRunner, tmp_path: Path):
```

Comment:
I don't have an opinion on the proper behavior here.  I went with the behavior recommended by `click` for [variadic arguments](https://click.palletsprojects.com/en/8.1.x/arguments/#variadic-arguments)

> ... we believe scripts should gracefully degrade into becoming noops if a variadic argument is empty. The reason for this is that very often, scripts are invoked with wildcard inputs from the command line and they should not error out if the wildcard is empty.